src/garage/experiment/local_runner.py

```diff
@@ -117,21 +124,28 @@ def obtain_samples(self, itr, batch_size=None):
                 This is a hint that the sampler may or may not respect.
 
         Returns:
-            One batch of samples.
+            list[dict]: One batch of samples.
 
         """
-        if self.train_args.n_epoch_cycles == 1:
+        if self._train_args.n_epoch_cycles == 1:
             logger.log('Obtaining samples...')
-        return self.sampler.obtain_samples(
-            itr, (batch_size or self.train_args.batch_size))
 
-    def save(self, epoch, paths=None):
+        paths = self.sampler.obtain_samples(
+            itr, (batch_size or self._train_args.batch_size))
+
+        for path in paths:
+            self.total_env_steps += len(path['rewards'])
+
+        return paths
+
+    def save(self, epoch):
         """Save snapshot of current batch.
 
         Args:
-            itr(int): Index of iteration (epoch).
-            paths(dict): Batch of samples after preprocessed. If None,
-                no paths will be logged to the snapshot.
+            epoch(int): Epoch.
+
+        Raises:
+            Exception: if save() is called before the runner is set up.
```

Comment:
please define a more specific exception class for this module. Using `Exception` means that clients have to `catch(Exception):` which is an antipattern.